zeppelin-plugins/launcher/k8s-standard/src/main/java/org/apache/zeppelin/interpreter/launcher/K8sRemoteInterpreterProcess.java

```diff
@@ -307,15 +307,19 @@ Properties getTemplateBindings(String userName) {
       // configure interpreter property "zeppelin.spark.uiWebUrl" if not defined, to enable spark ui through reverse proxy
       String webUrl = (String) properties.get("zeppelin.spark.uiWebUrl");
       if (StringUtils.isBlank(webUrl)) {
-        webUrl = "//{{PORT}}-{{SERVICE_NAME}}.{{SERVICE_DOMAIN}}";
+        webUrl = "{{PORT}}-{{SERVICE_NAME}}.{{SERVICE_DOMAIN}}";
+        String bindedWebUrl = sparkUiWebUrlFromTemplate(
+                webUrl,
+                webUiPort,
+                getPodName(),
+                k8sEnv.get(ENV_SERVICE_DOMAIN)
+        );
+        properties.put("zeppelin.spark.uiWebUrl", "//" + bindedWebUrl);
+
+        //configure the ingress.host to binded web url, to enable spark ui through ingress
+        k8sProperties.put("zeppelin.k8s.spark.ingress.host", bindedWebUrl);
       }
-      properties.put("zeppelin.spark.uiWebUrl",
```

Comment:
Why are you moving this part to the `if`?
In my environment I have defined a custom `webUrl` and I need the function `SparkUiWebUrlFromTemplate`.

Reply:
> Why are you moving this part to the `if`?
> In my environment I have defined a custom `webUrl` and I need the function `SparkUiWebUrlFromTemplate`.

Thanks for the comment. I realized that this is a mistake since the customed `zeppelin.spark.uiWebUrl` also needs to be bound by the function `SparkUiWebUrlFromTemplate`.